src/middlewared/middlewared/plugins/boot.py

```diff
@@ -112,8 +135,8 @@ async def attach(self, job, dev, options):
         # If the user is upgrading his disks, let's set expand to True to make sure that we
         # register the new disks capacity which increase the size of the pool
         await self.middleware.call('zfs.pool.online', BOOT_POOL_NAME, zfs_dev_part['name'], True)
-
         await self.update_initramfs()
+        await self.clear_disks_cache()
```

Comment:
If update-initramfs fails then the cache will not be cleared, even though pool already has other disks